flask_flatpages/page.py

```diff
@@ -63,13 +63,17 @@ def html(self):
     @cached_property
     def meta(self):
         """A dict of metadata parsed as YAML from the header of the file."""
-        meta = yaml.safe_load(self._meta)
+        meta = yaml.safe_load_all(self._meta)
         # YAML documents can be any type but we want a dict
         # eg. yaml.safe_load('') -> None
         #     yaml.safe_load('- 1\n- a') -> [1, 'a']
-        if not meta:
+
+        meta_list = list(meta)
+        meta_dict = dict(meta_list[0])
+
+        if not meta_dict:
             return {}
-        if not isinstance(meta, dict):
+        if not isinstance(meta_dict, dict):
```

Comment:
How can this condition fail?

Reply:
One scenario I can think of is the loaded YAML of the original `meta` object might not be properly formatted forcing the `dict` cast to fail. Exactly what was happening before (`---` at the end of YAML headers was throwing the exception) & made me open the PR.

Reply:
In that case, `ValueError` will be raised. `dict()` always returns a dict.